File: easy/challenge_03/Main.java
package code.eval.easy;

// Largest palindrome prime less than 1000
public class Challenge3 {

    public static void main(String[] args) {
        for (int i = 1000; i > 0; i--) {
            if (isPrime(i) && isPalindrome(i)) {
                System.out.println(i);
                System.exit(0);
            }
        }
    }

    private static boolean isPrime(int in) {
        if (in <= 1) { return false; }
        if (in == 2) { return true; }
        for (int i = 2 ; i < Math.sqrt(in) + 1 ; i++ ) {
            if (in % i == 0) { return false; }
        }
        return true;
    }

    private static boolean isPalindrome(int in) {
        String strIn = "" + in;
        for (int i = 0; i <= (strIn.length() / 2) ; i++ ) {
            if (strIn.charAt(i) != strIn.charAt(strIn.length() - (i + 1))) {
                return false;
            }
        }
        return true;
    }

}

File: easy/challenge_03/main.cpp
// Challenge 03:
// Larges prime palendrome less than 1000
#include <iostream>
#include <string>
#include <math.h>
using namespace std;

bool isPrime(int test) {
    switch(test) {
        case 1  : return false;
        case 2  : return true;
        default :
            // get rid of evens first
            if (test % 2 == 0) return false;

            // test only odd divisors in the loop
            for (int mod = 3; mod < sqrt(test) + 1 ; mod+=2) {
                if (test % mod == 0) return false;
            }
            return true;
    }
}

bool isPalindrome(int input) {
    std::string test = "" + input;
    for (int i = 0, j = test.length() - 1 ; i < j ; i++, j-- ) {
        if (test[i] != test[j])
            return false;
    }
    return true;
}

int main(int argc, char ** argv) {
    for (int i = 1000; i > 0 ; i++) {
        if (isPrime(i)) {
            if (isPalindrome(i)) {
                std::cout << i << std::endl;
            }
        }
    }
    return 0;
}
